pkg/security/registry/podsecuritypolicysubjectreview/rest.go

```diff
@@ -42,22 +43,22 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 	if !ok {
 		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a PodSecurityPolicySubjectReview: %#v", obj))
 	}
-
 	ns, ok := kapi.NamespaceFrom(ctx)
 	if !ok {
 		return nil, kapierrors.NewBadRequest("namespace parameter required.")
 	}
-
 	if errs := securityvalidation.ValidatePodSecurityPolicySubjectReview(pspsr); len(errs) > 0 {
 		return nil, kapierrors.NewInvalid(kapi.Kind("PodSecurityPolicySubjectReview"), "", errs)
 	}
-
-	userInfo := &user.DefaultInfo{Name: pspsr.Spec.User, Groups: pspsr.Spec.Groups}
+	groups := pspsr.Spec.Groups
+	if len(pspsr.Spec.User) > 0 { // if there's a user we mimic an authenticated user
+		groups = append(groups, bootstrappolicy.AuthenticatedGroup, bootstrappolicy.AuthenticatedOAuthGroup)
```

Comment:
This solves the problem only partially, since we're hardcoding some but not all actual user groups. I'll try to have a look into it... maybe I'll be more lucky in digging out what's wrong.